conans/test/integration/generators/test_custom_global_generators.py

```diff
@@ -0,0 +1,81 @@
+import os
+import textwrap
+
+from conans.test.assets.genconanfile import GenConanfile
+from conans.test.utils.tools import TestClient
+from conans.util.files import save
+
+
+def test_custom_global_generator():
+    c = TestClient()
+    generator = textwrap.dedent("""
+        class MyCustomGenerator:
+            def __init__(self, conanfile):
+                self._conanfile = conanfile
+            def generate(self):
+                pkg = self._conanfile.dependencies["pkg"].ref
+                self._conanfile.output.info(f"DEP: {pkg}!!")
+        """)
+    save(os.path.join(c.cache.custom_generators_path, "mygen.py"), generator)
+    conanfile = textwrap.dedent("""
+        [requires]
+        pkg/0.1
+
+        [generators]
+        MyCustomGenerator
+        """)
+    c.save({"pkg/conanfile.py": GenConanfile("pkg", "0.1"),
+            "conanfile.txt": conanfile})
+    c.run("create pkg")
+    c.run("install .")
+    assert "conanfile.txt: Generator 'MyCustomGenerator' calling 'generate()'" in c.out
+    assert "conanfile.txt: DEP: pkg/0.1!!" in c.out
+
+    # By CLI also works
+    conanfile = textwrap.dedent("""
+        [requires]
+        pkg/0.1
+        """)
+    c.save({"conanfile.txt": conanfile})
+    c.run("install . -g MyCustomGenerator")
```

Comment:
Hi @waruqi ,

It looks for a Python file in the $CONAN_HOME/extensions/generators folder and loads it dynamically if it contains a class definition that matches the name of the specified generator.

Reply:
Yes, exactly, they are not ``exported`` anymore, because they are not Conan versioned packages, and they are not addressed by ``requires = "something/version"`` (that can be achieved with the ``python_requires`` approach). They can be installed in the Conan home with ``conan config install <url/path>`` to make them available.

Reply:
> Not exactly, the folder structure need to be preserved, you can point to the upper folder, or you can use `conan config install` arguments `--source-folder` and `--target-folder` to correctly define the path. But in the home it should be inside the `extensions/generators` folder

Will this modify the user's global configuration path? But I don't want to change it, I just want to export my custom generator for use with conan install.

Can I always get $CONAN_HOME and copy my generators to $CONAN_HOME/extensions/generators?

Reply:
> Will this modify the user's global configuration path? But I don't want to change it, I just want to export my custom generator for use with conan install.

Yes, it is installed in the CONAN_HOME. But it will not affect users if not used. 

> Can I always get $CONAN_HOME and copy my generators to $CONAN_HOME/extensions/generators?

This is exactly what ``conan config install`` does.

If you are ``exporting``, you are creating a Conan package that must be ``required`` by some other package recipe. If you can add such a ``requires`` to some package recipe ``conanfile.py``, then you can already use the ``python_requires`` approach, it requires a ``conanfile.py`` and the ``generate()`` will take 2 lines more, but the operation principle is the same.

Reply:
ok, I see. thanks